### ISSUE TITLE
[BUG] LAG tests not working

### ISSUE THREAD

**Issue by ssasso** [ipspace/netlab#1689]
While trying to implement LAG in AOS-CX, I encountered this error in running the tests:

```
root@hippo:~/GIT_H/netlab/tests/integration/lag# netlab create 01-l3-lag.yml -p clab -d arubacx
Traceback (most recent call last):
  File "/usr/local/bin/netlab", line 7, in <module>
    exec(compile(f.read(), __file__, 'exec'))
  File "/root/GIT_H/netlab/netlab", line 14, in <module>
    netsim.cli.lab_commands(__file__)
  File "/root/GIT_H/netlab/netsim/cli/__init__.py", line 353, in lab_commands
    mod.run(sys.argv[arg_start:])   # type: ignore
  File "/root/GIT_H/netlab/netsim/cli/create.py", line 81, in run
    augment.main.transform(topology)
  File "/root/GIT_H/netlab/netsim/augment/main.py", line 109, in transform
    transform_data(topology)
  File "/root/GIT_H/netlab/netsim/augment/main.py", line 66, in transform_data
    modules.pre_transform(topology)
  File "/root/GIT_H/netlab/netsim/modules/__init__.py", line 77, in pre_transform
    module_transform("pre_transform",topology)
  File "/root/GIT_H/netlab/netsim/modules/__init__.py", line 462, in module_transform
    mod_load[m].call("module_"+method,topology)
  File "/root/GIT_H/netlab/netsim/utils/callback.py", line 42, in call
    return method(*args, **kwargs)
  File "/root/GIT_H/netlab/netsim/modules/lag.py", line 372, in module_pre_transform
    process_lag_links(topology)             # Expand lag.members into additional p2p links
  File "/root/GIT_H/netlab/netsim/modules/lag.py", line 324, in process_lag_links
    create_lag_member_links(l,topology)
  File "/root/GIT_H/netlab/netsim/modules/lag.py", line 178, in create_lag_member_links
    def analyze_lag(members: list, node_count: dict) -> tuple[bool,bool,str]:
TypeError: 'type' object is not subscriptable
```

(note: same happens using device type 'eos', so this is not device dependant)

Same for test *02*:

```
root@hippo:~/GIT_H/netlab/tests/integration/lag# netlab create 02-lag-vlan-trunk.yml -p clab -d eos
Traceback (most recent call last):
  File "/usr/local/bin/netlab", line 7, in <module>
    exec(compile(f.read(), __file__, 'exec'))
  File "/root/GIT_H/netlab/netlab", line 14, in <module>
    netsim.cli.lab_commands(__file__)
  File "/root/GIT_H/netlab/netsim/cli/__init__.py", line 353, in lab_commands
    mod.run(sys.argv[arg_start:])   # type: ignore
  File "/root/GIT_H/netlab/netsim/cli/create.py", line 81, in run
    augment.main.transform(topology)
  File "/root/GIT_H/netlab/netsim/augment/main.py", line 109, in transform
    transform_data(topology)
  File "/root/GIT_H/netlab/netsim/augment/main.py", line 66, in transform_data
    modules.pre_transform(topology)
  File "/root/GIT_H/netlab/netsim/modules/__init__.py", line 77, in pre_transform
    module_transform("pre_transform",topology)
  File "/root/GIT_H/netlab/netsim/modules/__init__.py", line 462, in module_transform
    mod_load[m].call("module_"+method,topology)
  File "/root/GIT_H/netlab/netsim/utils/callback.py", line 42, in call
    return method(*args, **kwargs)
  File "/root/GIT_H/netlab/netsim/modules/lag.py", line 372, in module_pre_transform
    process_lag_links(topology)             # Expand lag.members into additional p2p links
  File "/root/GIT_H/netlab/netsim/modules/lag.py", line 324, in process_lag_links
    create_lag_member_links(l,topology)
  File "/root/GIT_H/netlab/netsim/modules/lag.py", line 178, in create_lag_member_links
    def analyze_lag(members: list, node_count: dict) -> tuple[bool,bool,str]:
TypeError: 'type' object is not subscriptable
```

**Comment by ssasso** [comment]
(running on `Python 3.8.10` on `Ubuntu 20.04.6 LTS`)

**Comment by jbemmel** [comment]
See https://github.com/ipspace/netlab/blob/ed975091c22890195d8554f8101a25552452df7d/.github/workflows/tests.yml#L14

Python 3.8 went end-of-support on 2024-10-7, minimal supported version is 3.9

Should add a check for that though

**Comment by ssasso** [comment]
But this goes totally against `netlab is a Python3 package[1](https://netlab.tools/install/#nsl) that runs on Python 3.8 or later`

I don't think we want to say 'if you have ubuntu 20.04 then we don't care', isn't it? @ipspace

**Comment by ssasso** [comment]
or, at least, if we want to deprecate python 3.8 for netlab - I could be ok with it - but then it must be deprecated for everything, not only for a single module. on the other way, if we don't want to deprecate it, I think we must be sure every module works with 3.8

**Comment by ssasso** [comment]
https://github.com/ipspace/netlab/blob/ed975091c22890195d8554f8101a25552452df7d/setup.py#L40

**Comment by jbemmel** [comment]
> or, at least, if we want to deprecate python 3.8 for netlab - I could be ok with it - but then it must be deprecated for everything, not only for a single module. on the other way, if we don't want to deprecate it, I think we must be sure every module works with 3.8

Your bug report illustrates why we needed to deprecate 3.8 support - but there were some loose ends to be cleaned up, see PR

**Comment by ipspace** [comment]
> See
> 
> https://github.com/ipspace/netlab/blob/ed975091c22890195d8554f8101a25552452df7d/.github/workflows/tests.yml#L14
> 
> Python 3.8 went end-of-support on 2024-10-7, minimal supported version is 3.9

How about fixing your stuff? It took me less than 30 seconds to find https://stackoverflow.com/questions/75202610/typeerror-type-object-is-not-subscriptable-python when googling for the exact error.

Also, you just proved why I need to add Python 3.8 back to that list.

**Comment by ipspace** [comment]
> I don't think we want to say 'if you have ubuntu 20.04 then we don't care', isn't it? @ipspace

Not for another week or two ;) I'm guessing that at least some of our audience hates unnecessary upgrades that break half of the stuff that used to work flawlessly as much as I do.

**Comment by ipspace** [comment]
The code is fixed in 4c31a0f59b48b4438163c563bd105d021256d173 and 379026a2127f2512d6c9b1ecc1d4cbcaa91fcb1b. Probably took me an order of magnitude less time than the "but we really should not support Python 3.8 anymore" bikeshedding.
